morpheus/stages/input/file_source.py

```diff
@@ -0,0 +1,319 @@
+        Supported extensions: 'csv', 'json', 'jsonlines' and 'parquet'.
+    repeat : int, default = 1, min = 1
+        Repeats the input dataset multiple times. Useful to extend small datasets for debugging.
+    filter_null : bool, default = True
+        Whether or not to filter rows with a null 'data' column. Null values in the 'data' column can cause issues down
+        the line with processing. Setting this to True is recommended.
+    parser_kwargs : dict, default = {}
+        Extra options to pass to the file parser.
+    """
+
+    def __init__(self,
+                 config: Config,
+                 files: typing.List[str],
+                 watch: bool = False,
+                 watch_interval: float = 1.0,
+                 sort_glob: bool = False,
+                 recursive: bool = True,
+                 queue_max_size: int = 128,
+                 batch_timeout: float = 5.0,
+                 file_type: FileTypes = FileTypes.Auto,
+                 repeat: int = 1,
+                 filter_null: bool = True,
+                 parser_kwargs: dict = None):
+
+        super().__init__(config)
+
+        if not files:
+            raise ValueError("The 'files' cannot be empty.")
+
+        if watch and len(files) != 1:
+            raise ValueError("When 'watch' is True, the 'files' should contain exactly one file path.")
+
+        self._files = list(files)
+        self._watch = watch
+        self._sort_glob = sort_glob
+        self._recursive = recursive
+        self._queue_max_size = queue_max_size
+        self._batch_timeout = batch_timeout
+        self._file_type = file_type
+        self._filter_null = filter_null
+        self._parser_kwargs = parser_kwargs or {}
+        self._watch_interval = watch_interval
+        self._repeat_count = repeat
+
+    @property
+    def name(self) -> str:
+        """Return the name of the stage"""
+        return "file-source"
+
+    def supports_cpp_node(self) -> bool:
+        """Indicates whether or not this stage supports a C++ node"""
+        return False
+
+    def _has_remote_paths(self) -> bool:
+        return any(urlsplit(file).scheme for file in self._files if "://" in file)
+
+    def _build_source(self, builder: mrc.Builder) -> StreamPair:
+
+        if self._build_cpp_node():
+            raise RuntimeError("Does not support C++ nodes")
+
+        if self._watch and not self._has_remote_paths():
+            input_glob = self._files[0]
+            watcher = DirectoryWatcher(
+                input_glob=input_glob,
+                watch_directory=self._watch,
+                max_files=None,  # This is not being used in the latest version.
+                sort_glob=self._sort_glob,
+                recursive=self._recursive,
+                queue_max_size=self._queue_max_size,
+                batch_timeout=self._batch_timeout)
+
+            out_stream = watcher.build_node(self.unique_name, builder)
+            out_type = list[str]
```

Comment:
Question Is `DirectoryWatcher` faster than `fsspec`? Why would we use it over `fsspec`?

Reply:
I have used the `DirectoryWatcher` since it is mentioned here in the [issue](https://github.com/nv-morpheus/Morpheus/issues/976). We can just use the `fsspec`